src/extremeweatherbench/metrics.py

```diff
@@ -103,73 +126,108 @@ class MaxOfMinTempMAE(Metric):
     """Mean absolute error of forecasted highest minimum temperature values."""
 
     def compute(self, forecast: xr.DataArray, observation: xr.DataArray):
-        max_min_mae_values = []
-        observation_spatial_mean = observation.mean(["latitude", "longitude"])
-        observation_spatial_mean = observation_spatial_mean.where(
-            observation_spatial_mean.time.dt.hour % 6 == 0, drop=True
-        )
-        forecast_spatial_mean = forecast.mean(["latitude", "longitude"])
-        for init_time in forecast_spatial_mean.init_time:
-            if forecast_spatial_mean.name == "air_temperature":
-                # Keep only times at 00, 06, 12, and 18Z
-                # Group by dayofyear and check if each day has all 4 synoptic times
-                valid_days = (
-                    observation_spatial_mean.groupby("time.dayofyear").count("time")
-                    == 4
-                )
-                # Only keep days that have all 4 synoptic times
-                observation_spatial_mean = observation_spatial_mean.where(
-                    observation_spatial_mean.time.dt.dayofyear.isin(
-                        valid_days.where(valid_days).dropna(dim="dayofyear").dayofyear
-                    ),
-                    drop=True,
-                )
-                max_min_timestamp = observation_spatial_mean.where(
-                    (
-                        observation_spatial_mean
-                        == observation_spatial_mean.groupby("time.dayofyear")
-                        .min()
-                        .max()
-                    ),
-                    drop=True,
-                ).time
-                max_min_value = observation_spatial_mean.sel(
-                    time=max_min_timestamp
-                ).values
-                init_forecast_spatial_mean, _ = self.align_datasets(
+        if forecast.name == "air_temperature":
```

Comment:
Same comment as above re: filtering based on the `forecast.name`. Just get rid of it; let users be responsible for using these methods appropriately.